output/src/main/java/com/scottlogic/deg/output/writer/sql/SqlDataSetWriter.java

```diff
@@ -0,0 +1,99 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.scottlogic.deg.output.writer.sql;
+
+import com.scottlogic.deg.common.output.GeneratedObject;
+import com.scottlogic.deg.common.profile.Field;
+import com.scottlogic.deg.common.profile.ProfileFields;
+import com.scottlogic.deg.output.writer.DataSetWriter;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.math.BigDecimal;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+class SqlDataSetWriter implements DataSetWriter {
+    private static final DateTimeFormatter standardDateFormat = DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+    private final Writer writer;
+    private final ProfileFields fieldOrder;
+    private final String tableName;
+
+    SqlDataSetWriter(Writer writer, ProfileFields fieldOrder, String tableName) {
+        this.writer = writer;
+        this.fieldOrder = fieldOrder;
+        this.tableName = tableName;
+    }
+
+    @Override
+    public void writeRow(GeneratedObject row) throws IOException {
+        // eg insert into dbo.Currencies (currency, from_date, to_date) values
+        //    ( 'YYY', '1762-02-16T00:00:00Z', '1980-02-16T00:00:00Z');
+
+        Optional<String> columnNamesCsvString = reduceStringStreamToCsvString(
+            fieldOrder.stream()
+                .map(Field::toString)
+        );
+        Optional<String> columnValuesCsvString = reduceStringStreamToCsvString(
+            fieldOrder.stream()
+                .map(row::getFormattedValue)
+                .map(SqlDataSetWriter::fieldValueToString)
+        );
+
+        if (!columnNamesCsvString.isPresent() || !columnValuesCsvString.isPresent()) {
+            return;
+        }
+
+        writer
+            .append("INSERT INTO ")
+            .append(tableName).append(" (")
+            .append(columnNamesCsvString.get())
+            .append(") VALUES (")
+            .append(columnValuesCsvString.get())
+            .append(");\n")
+            .flush();
+    }
+
+    @Override
+    public void close() throws IOException {
+        writer.close();
+    }
+
+    private static String fieldValueToString(Object value) {
+        if (value == null) {
+            return "NULL";
+        }
+        else if (value instanceof BigDecimal) {
+            return ((BigDecimal) value).toPlainString();
+        }
+        else if (value instanceof OffsetDateTime) {
+            return "'" + standardDateFormat.format((OffsetDateTime) value) + "'";
+        }
+        else if (value instanceof String) {
+            return "'" + value + "'";
+        }
+        else {
+            return value.toString();
+        }
+    }
+
+    private static Optional<String> reduceStringStreamToCsvString(Stream<String> stream) {
+        return stream.reduce((t, u) -> t + ", " + u);
```

Comment:
Do we want `", "` (with space) or `","` (without space) for writing to CSV?

Reply:
renamed method so viewer of the code is not expecting normal CSV string.  We do want the space before the comma as it reads nicer within the insert statement produced